test/toolTests.ts

```diff
@@ -81,24 +81,83 @@ describe('Tool Tests', function () {
         });
     });
 
-    it('has status code in exception dictionary for non 200 response', function () {
-        this.timeout(2000);
 
-        return new Promise<void>(async(resolve, reject)=> {
-            try {
-                let nonExistentUrl: string = "http://httpbin.org/nonexistent";
-                let downPath: string = await toolLib.downloadTool(nonExistentUrl);
+    
+    describe('has status code in exception dictionary for HTTP error code responses', function () {
```

Comment:
I think the only test we need is a download that redirects (once) and we make sure it's 200.  The tests below are the responsibility of the typed rest client